sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2270,4 +2270,15 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     val mapWithBinaryKey = map(lit(Array[Byte](1.toByte)), lit(1))
     checkAnswer(spark.range(1).select(mapWithBinaryKey.getItem(Array[Byte](1.toByte))), Row(1))
   }
+
+  test("SPARK-24556: ReusedExchange should rewrite output partitioning for RangePartitioning") {
```

Comment:
this is not an end-to-end test, let's put it in `PlannerSuite` and also test cached table.

Reply:
please also mention cached table in PR title